Fix custom meter error with zone mechanical ventilation and airloop variables

LipingWang opened the pull request:
Pull request overview
---------------------
 - Fixes #7655 
 - Fixed the issue with custom meter.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

LipingWang committed:
Fixed the custom meter issue

https://github.com/NREL/EnergyPlus/issues/7655
The issue was also reported in the helpdesk ticket# 15343.
The function GetCustomMeterInput (), which processes the output variables for custom meters, ran before the function AllocateAndSetUpVentReports(), which sets up many output variables.
The issue was fixed by moving the function UpdateMeterReporting() after the function CreateEnergyReportStructure().

LipingWang committed:
Fixed the custom meter issue

NREL#7655
The issue was also reported in the helpdesk ticket# 15343.
The function GetCustomMeterInput (), which processes the output variables for custom meters, ran before the function AllocateAndSetUpVentReports(), which sets up many output variables.
The issue was fixed by moving the function UpdateMeterReporting(), together with the function verifyCustomMetersElecPowerMgr(), after the function CreateEnergyReportStructure().

LipingWang committed:
Merge branch 'develop' into error_custom_meter

Myoldmopar commented:
New warnings showing up in the error files with this change.  It looks like there are side effects to making this change.

mjwitte commented:
Timing is everything during initial setup. Will rethink this - we may need to split up what `CreateEnergyReportStructure` does so that the output variables are present when `UpdateMeterReporting` is called.

LipingWang committed:
Fixed the issue for custom meter

Move the function AllocateAndSetUpVentReports() before  UpdateMeterReporting().

LipingWang committed:
Merge branch 'develop' into error_custom_meter

LipingWang committed:
Fix the issue with custom meter

Move the function AllocateAndSetUpVentReports() before the function UpdateMeterReporting().

mjwitte commented on src/EnergyPlus/SimulationManager.cc:
```diff
@@ -281,6 +281,7 @@ namespace SimulationManager {
         using SizingManager::ManageSizing;
         using SystemReports::CreateEnergyReportStructure;
         using SystemReports::ReportAirLoopConnections;
+        using SystemReports::AllocateAndSetUpVentReports;
```
@LipingWang Minor detail. We're trying to move away from `using` statements and be explicit when actually using the function or variable. So, when you add the unit test, please delete this line.

mjwitte commented on src/EnergyPlus/SimulationManager.cc:
```diff
@@ -410,12 +411,12 @@ namespace SimulationManager {
             SetupNodeVarsForReporting();
             MetersHaveBeenInitialized = true;
             SetupPollutionMeterReporting();
+            AllocateAndSetUpVentReports();
```
And here change to `SystemReports::AllocateAndSetUpVentReports();`

mjwitte commented on src/EnergyPlus/SimulationManager.cc:
```diff
@@ -428,7 +429,7 @@ namespace SimulationManager {
             CheckControllerLists(ErrFound);
             if (ErrFound) TerminalError = true;
-
+        
```
Very minor - remove trailing spaces.

mjwitte reviewed (commented):
This looks good. Now it needs a new unit test. `OutputProcessor.unit.cc:: TEST_F(EnergyPlusFixture, OutputProcessor_DuplicateMeterCustom)` would be a good starting point.

LipingWang committed:
Merge branch 'develop' into error_custom_meter

LipingWang committed:
Fix the issue with custom meter

Fixes #7655
Added a unit test for Meter:Custom TEST_F(EnergyPlusFixture, OutputProcessor_MeterCustomSystemEnergy).

mitchute committed:
Merge branch 'develop' into error_custom_meter

mitchute committed:
minor cleanup to remove stray comments. apply style to OutputProcessor.unit.cc

mitchute committed:
cleanup whitespace

mitchute commented:
@mjwitte it looks like you've already been through this. I did build and run the unit tests locally, and it looks good. I also went ahead and ran clang-format on ```OutputProcessors.unit.cc``` but I probably won't do that again since it muddles up the changelog. FFR the unit test ```OutputProcessor_MeterCustomSystemEnergy``` was added during this PR.

I'll let CI run again, but this should be ready to merge.

mjwitte commented:
@mitchute I helped @LipingWang develop this fix, so someone else (like you) should be doing the full review on this.

mitchute commented:
Looks good. Merging.

mitchute commented:
@LipingWang @mjwitte this looks good. I just ran the defect IDF file [(here)](https://github.com/NREL/EnergyPlusDevSupport/blob/54a4c4d817dfed8cffec1be0ff57ba6908baf549/DefectFiles/7000s/7655/5ZoneFanCoilDOASCool_test_meter_output_out.idf) and verified that the extra error file warning messages are now gone. Merging.
